src/librustc_session/session.rs

```diff
@@ -491,11 +521,14 @@ impl Session {
     /// dependency tracking. Use tcx.features() instead.
     #[inline]
     pub fn features_untracked(&self) -> &rustc_feature::Features {
-        self.features.get()
+        self.features.get().unwrap()
     }
 
     pub fn init_features(&self, features: rustc_feature::Features) {
-        self.features.set(features);
+        match self.features.set(features) {
+            Ok(()) => {}
+            Err(_) => panic!("`features` was initialized twice"),
```

Comment:
Let's use expect here like in init_crate_types.

Reply:
`Features` doesn't implement `Debug`, and I suspect a generated one would be quite large. Still want me to do this? I could also just `unwrap` here.

Reply:
Ah okay, I didn't realize that was the reason. Seems fine to leave as-is then